test/index.js

```diff
@@ -764,6 +777,11 @@ test('traverse', function() {
   eq(Z.traverse(Identity, identInc, [1, 2, 3]), Identity([2, 3, 4]));
   eq(Z.traverse(Identity, identInc, Nil), Identity(Nil));
   eq(Z.traverse(Identity, identInc, Cons(1, Cons(2, Cons(3, Nil)))), Identity(Cons(2, Cons(3, Cons(4, Nil)))));
+
+  assert.doesNotThrow(function() {
+    // traverse and run the accumulator. this should stack safe.
```

Comment:
And as I understand it's not quite stack safe, just usage of stack is logarithmic. so I think it would  be better to be much precise here, also it would be nice to state this in description of the Array.traverse.